src/System.Private.Xml/tests/XmlReader/Tests/AsyncReaderLateInitTests.cs

```diff
@@ -68,23 +68,45 @@ public static void ReadAfterInitializationWithTextReaderOnAsyncReaderDoesNotThro
 
         [Fact]
         [SkipOnTargetFramework(TargetFrameworkMonikers.Uap |  //[ActiveIssue(13121)]  // Path cannot be resolved in UWP
-        TargetFrameworkMonikers.NetFramework)]    // Full framework uses XmlUrlResolver instead of SystemPathResolver (which doesn't allow access to this path)
+        TargetFrameworkMonikers.NetFramework)]  //[ActiveIssue(19806)] // Full framework uses WebRequest instead of HttpRequest
         public static void ReadAsyncAfterInitializationWithUriThrows()
         {
             using (XmlReader reader = XmlReader.Create("http://test.test/test.html", new XmlReaderSettings() { Async = true }))
             {
-                Assert.Throws<XmlException>(() => reader.ReadAsync().GetAwaiter().GetResult());
+                Assert.Throws<System.Net.Http.HttpRequestException>(() => reader.ReadAsync().GetAwaiter().GetResult());
+            }
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.NetFramework)] // Full framework uses WebRequest instead of HttpRequest
+        public static void ReadAsyncAfterInitializationWithUriThrows_FullFramework()
+        {
+            using (XmlReader reader = XmlReader.Create("http://test.test/test.html", new XmlReaderSettings() { Async = true }))
+            {
+                Assert.Throws<System.Net.WebException>(() => reader.ReadAsync().GetAwaiter().GetResult());
```

Comment:
https://github.com/dotnet/corefx/blob/master/src/Common/tests/System/AssertExtensions.cs#L63

Perhaps we should add an extension that doesn't expect a paramName